### PR TITLE
More improvements to Tileset.md

### PR DESCRIPTION
**Pull request by Eauix**
As I've been creating and experimenting with tilesets for a long while now, I've decided to contribute to the tileset.md page, adding a more detailed explanation on how tile attributes work, which attributes are available and why some attributes are the number they have. I also added a guide on how to create special object tiles like coins, bonus blocks and bricks, as the attributes that made these behavior no longer work.

### CODE REVIEW

File: Tileset.md [hunk 1]
```diff
@@ -68,16 +76,32 @@ It is also possible to extract parts of bigger images to create tiles:
...
     0 0 0 0
-    1 1 1 0)
+    1 1 0 0)
   (image "tiles/[tilegroup]/[multipletiles].png")
 )
 ```
 
 ---
 
+For a tile that looks one way in editor and other way in game, you can set the images separately:
```
Comment (Vankata453) [inline Tileset.md:89]:
```suggestion
For a tile which looks different in the editor, you can set the images separately:
```

File: Tileset.md [hunk 2]
```diff
@@ -68,16 +76,32 @@ It is also possible to extract parts of bigger images to create tiles:
...
+(tiles
+  (width 1)
+  (height 1)
+  (ids 9)
+  (attributes 3)
+  (images "tiles/[tilegroup]/[tile].png")
+  (editor-images "tiles/[tilegroup]/[tile]-editor.png")
+)
+```
+If you want the tile to be invisible in game, you just remove the "(images)" section
```
Comment (Vankata453) [inline Tileset.md:101]:
```suggestion
If you want the tile to be invisible in-game, remove the `(images)` section.
```

File: Tileset.md [hunk 3]
```diff
@@ -315,6 +376,116 @@ tilegroup.
...
+      "objects/[object]/[object].png"
+    )
+    (object-name "[object class name]")
+    (object-data "
+      ([propriety1] 50)
+      ([propriety2] \"[string]\")
+    ")
+)
+```
+Be aware that any object propriety that has quotation marks on it, must get a backslash (\\) before any quotation marks, as they are quotation marks inside other quotation marks.
```
Comment (Vankata453) [inline Tileset.md:403]:
```suggestion
Be aware that any quotation mark in `object-data` must be escaped by adding a backslash (`\`) before it, as it's a quotation mark inside other quotation marks.
```

File: Tileset.md [hunk 4]
```diff
@@ -315,6 +376,116 @@ tilegroup.
...
+
+---
+If you find it difficult to write the object proprieties you want to add as a tile, you can place this object in your level, open the level in a text editor and copy the information you'll find about this object, for example, if you want a Mr.Snowball, you get in the level file and search for it, you'll find:
```
Comment (Vankata453) [inline Tileset.md:406]:
```suggestion
If you find it difficult to write the object proprieties you want to add as a tile, you can place the object in your level, open the level in a text editor and copy the properties present for the object. For example, if you add a Mr.Snowball to a level and customize it, in the level file you'll find:
```

File: Tileset.md [hunk 5]
```diff
@@ -315,6 +376,116 @@ tilegroup.
...
+
+```
+(snowball
+      (z-pos 52)
+      (direction "left")
+      (x 288)
+      (y 128)
+)
+```
+From this information, you may extract what you want to turn into a tile, you must ignore the x and y positions as they will be defined as where the tile is.
```
Comment (Vankata453) [inline Tileset.md:416]:
```suggestion
From this information, you may extract what you want to turn into a tile. You must ignore the `x` and `y` position properties as they will be defined, depending on where the tile is.
```

File: Tileset.md [hunk 6]
```diff
@@ -315,6 +376,116 @@ tilegroup.
...
+    )
+    (object-name "trampoline")
+    (object-data "
+      (type \"stationary\")
+      (z-pos 50)
+    ")
+)
+```
+---
+Now let's be creative, let's say you want to add a bonus block that is orange and has 2 Mr.Icecubes inside it:
```
Comment (Vankata453) [inline Tileset.md:463]:
```suggestion
Now let's be creative, let's say you want to add a bonus block that is orange and has 2 Mr.Iceblocks inside it:
```

**Review by provigz — COMMENTED** [review]
Additionally, maybe we should also add info about the relatively new `deprecated` attribute for `(tiles)`.

**Comment by Eauix** [comment]
> Additionally, maybe we should also add info about the relatively new `deprecated` attribute for `(tiles)`.

Done

**Comment by Vankata453** [comment]
By the way, you can use "Add suggestions to batch" from the "Files" tab to not have to commit each suggestion separately.

File: Tileset.md [hunk 7]
```diff
@@ -310,12 +379,153 @@ tilegroup.
...
+    ")
+)
+```
+Again, if you feel confused with what information you must type in the object name and data sections, you can copy them from an object that's already placed in your level file!
+<hr>
+
+Deprecated Tiles
+================
+
+When a tile becomes outdated due to it being replaced by a different tile / object / decal - or when it is no longer necessary for level design, this tile is usually made deprecated for the sake of compatibility with older levels that were built with them before. Deprecated tiles can still be loaded by the level and will work as they used to, but they're not available to build with, nor can they be copied when editing the level.
```
Comment (Vankata453) [inline Tileset.md:500]:
```suggestion
When a tile becomes outdated due to it being replaced by a different tile / object / decal, or when it is no longer necessary for level design, it is usually made deprecated for the sake of compatibility with older levels which include it. Deprecated tiles can still be loaded by the level and will work as they used to, but they cannot be added to tilegroups, nor can they be copied when editing the level.
```

File: images/aatriangletypes2.png
Comment (Vankata453) [inline images/aatriangletypes2.png:1]: I believe the text can be improved too, just like the numbers?

File: Tileset.md [hunk 8]
```diff
@@ -7,12 +7,20 @@ These tilesets are included into the level by a `tiles` and `tilegroup` entry.
...
    * [Adding Tilegroups](#step-3-adding-tilegroups)
+5. [Special Tiles](#special-tiles)
+   * [Adding Objects As Tiles](#adding-objects-as-tiles)
+6. [Deprecated Tiles](#deprecated-tiles)
+   * [Deprecating Your Old Tiles](#deprecating-your-old-tiles)
+
+<hr>
 
 Introduction
 ============
```
Comment (Vankata453) [inline Tileset.md:26]: Under the "Introduction" -> "Examples" section, all separators are thicker. I believe they should be thinner as to not be mixed up with the ones separating the main parts of the document (1-6).

Comment (Eauix) [reply Tileset.md:26]: Turns out there's no way of creating a thin line in MD without adding a title. I tried `<hr style="height:1px;">` but it doesn't seem to support CSS. I also tried to add a title with an invisible character, it made the line but created a really large empty space before the line, so, not sure on what to do about this

**Pull request merged**